internal/request.go

```diff
@@ -50,7 +50,15 @@ func (s *SPOA) processRequest(msg spoe.Message) ([]spoe.Action, error) {
 			var ok bool
 			app, ok = s.applications[arg.Value.(string)]
 			if !ok {
-				return nil, fmt.Errorf("application %q not found", arg.Value.(string))
+				if len(s.defaultApplication) > 0 {
+					fmt.Printf("application %q not found, using default application %q\n", arg.Value.(string), s.defaultApplication)
+					app, ok = s.applications[s.defaultApplication]
+					if !ok {
+						return nil, fmt.Errorf("default application %q not found", s.defaultApplication)
+					}
+				} else {
+					return nil, fmt.Errorf("application %q not found", arg.Value.(string))
+				}
```

Comment:
Better if using the parameter in the end of the sentence, without type casting, which follows the component patterns. E.g.:

```
    return nil, fmt.Errorf("default application not found: %s", s.defaultApplication)
...
    return nil, fmt.Errorf("application not found: %v", arg.Value)
```

Reply:
Thanks for suggestion. Updated error messages.